run_dir/static/js/data_del_plot.js

```diff
@@ -0,0 +1,360 @@
+            series_name = data[d].delivery;
+            view_color = view_coloring(series_name);
+        }else if (view_type == 'sample'){
+            if (data[d].sample.includes('Finished')){
+                series_name = "Finished Library";
+            }else if (data[d].sample.includes('total')){
+                series_name = "total RNA";
+            }else if (data[d].sample.includes('Tissue')){
+                series_name = "Tissue";
+            }else if (data[d].sample.includes('Genomic')){
+                series_name = "Genomic DNA";
+            }else if (data[d].sample.includes('Amplicon')){
+                series_name = "Amplicon";
+            }else{
+                series_name = "Other";
+            }
+            view_color = view_coloring(series_name);
+        }else if (view_type == 'bp'){
+            if (data[d].bp == 'Yes'){
+                series_name = "BP bioinformatics";
+            }else if (data[d].bp == 'No'){
+                series_name = "No BP";
+            }else if (data[d].bp == 'Special'){
+                series_name = "Special";
+            }else{
+                series_name = "Other";
+            }
+            view_color = view_coloring(series_name);
+            }
+        if (!series.hasOwnProperty(series_name)){
+            series[series_name] = {
+                step: true,
+                name: series_name,
+                color: view_color,
+                data: [],
+            };
+            series.length += 1; 
+        }else if (view_type == 'cumulative'){
+            cumul_xaxis = "Total filesize delivered";
+            dp = {
+                y: fs_sum,
+                data: fs_sum
+            };
+            series[series_name].data.push(dp);
+            categories.push(cumul_xaxis);
+        }else{ 
+            dp = {
+                y: data[d][key],
+                name: project_name + ': ' + "<i>" + date_del + "</i>",
+                ownURL: bioinfo_link
+            };
+            series[series_name].data.push(dp);
+            categories.push(date_del);
+      }
+      var proper_series = []
+      for (s in series) {
+          proper_series.push(series[s]);
+      }
+    }
+  return [categories, proper_series];
+}
+
+function get_plot_data(key, name, view_type, search_string="", filter_inst_type){
+    $.getJSON('/api/v1/proj_staged/'+search_string, function(data) {
+        window.current_plot_data=data;
+        update_color_schemes(data);
+        make_plot(key, name, view_type, filter_inst_type);
+    });
+}
+//Since we have been confused by the coloring in the past, I set a few default colors to re-use for simplicity.
+function view_coloring(series_name){
+    switch(series_name){
+        case "All projects":
+            return current_color_schemes[1].hex();
+        case "NovaSeq":
+            return current_color_schemes[0].hex();
+        case "MiSeq":
+            return current_color_schemes[1].hex();
```

Comment:
You can collapse cases if they have the same result like 
 ```
 case "All projects":
 case "MiSeq":
 case "WG-reseq":
 case ...
         return current_color_schemes[1].hex();
```

Reply:
Nice! It makes the code shorter! ^^